src/tests/ftest/recovery/pass_2.py

```diff
@@ -0,0 +1,211 @@
+
+        # 2. Call dmg storage query usage to store the default space utilization.
+        rank_to_free_orig = self.get_rank_to_free()
+        self.log.info("rank_to_free_orig = %s", rank_to_free_orig)
+
+        # 3. Prepare to copy the pool path before stopping servers.
+
+        # In order to copy the pool directory without password, there are several things
+        # to determine and set up.
+
+        # 3-1. Determine source host and destination host. Source host is where rank 0 is.
+        # Destination host is the other host.
+        rank_to_host = self.server_managers[0].ranks
+        self.log.info("rank_to_host = %s", rank_to_host)
+        src_host = NodeSet(rank_to_host[0])
+        dst_host = None
+        for rank in range(1, 4):
+            if rank_to_host[rank] != str(src_host):
+                dst_host = NodeSet(rank_to_host[rank])
+                break
+        self.log.info("src_host = %s; dst_host = %s", src_host, dst_host)
+
+        # 3-2. Determine source and destination mount point. First, find the source mount
+        # point that maps to rank 0. Then use the same mount point for the destination.
+        # This way, we can handle the case where the mount point name is changed in the
+        # future. At the same time, determine the destination rank, which is where
+        # dst_mount is mapped.
+        # dst_mount = "/mnt/daos0"
+        src_mount = None
+        dst_mount = None
+        dst_rank = None
+        host_list = str(self.hostlist_servers)
+        dmg_command = self.get_dmg_command()
+        storage_query_out = dmg_command.storage_query_usage(
+            host_list=host_list)
+        hash_dict = storage_query_out["response"]["HostStorage"]
+        for storage_dict in hash_dict.values():
+            if str(src_host) in storage_dict["hosts"]:
+                # Determine source mount point that maps to rank 0.
+                for scm_namespace in storage_dict["storage"]["scm_namespaces"]:
+                    if scm_namespace["mount"]["rank"] == 0:
+                        src_mount = scm_namespace["mount"]["path"]
+                        # Use the same mount point as source.
+                        dst_mount = src_mount
+        for storage_dict in hash_dict.values():
+            if str(dst_host) in storage_dict["hosts"]:
+                # Determine destination rank that mpas to dst_mount.
+                for scm_namespace in storage_dict["storage"]["scm_namespaces"]:
+                    if scm_namespace["mount"]["path"] == dst_mount:
+                        dst_rank = scm_namespace["mount"]["rank"]
+
+        # 4. Stop servers.
+        dmg_command.system_stop()
+
+        # 5. Copy /mnt/daos?/<pool_path> from the engine where we created the pool to
+        # another engine where we didn’t create.
+
+        # 5-1. Since we're running rsync as user, update the mode of the source pool
+        # directory to 777.
+        self.log.info("Update mode of the source pool directory.")
```

Comment:
FYI that `self.log_step` is useful for this, but you'd want to use it for all steps in this test

Reply:
Thanks for the suggestion. I looked at `self.log_step` and it automatically increments the step value to print from 1, but I want to print 5-1 instead of 1. Also, I don't really see the need of printing every step for this test. I'm printing the descriptions about the commands to make things clearer during step 5 because I'm issuing a bunch of commands in a sequence.

Or you may think that we should print every test step with `self.log_step` and that also makes sense.

Reply:
> Or you may think that we should print every test step with self.log_step and that also makes sense

For `log_step` to make sense, you'd need to use it with every step. Anyway, this is new, and we have some kinks to work out, so not required to use at this point